### PR TITLE
Feature important information UI 

### PR DESCRIPTION
**Pull request by Juliehoejrisp**
Made information banner UI for iOS and connected to Backend with link.

### CODE REVIEW

File: NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs [hunk 1]
```diff
@@ -281,6 +309,9 @@ void SetOnOffBtnState(bool isRunning)
 
         void SetupStyling()
         {
+            InformationBannerLbl.Font = StyleUtil.Font(StyleUtil.FontType.FontMedium, 18, 22);
+            InformationBannerLbl.Text = "TODO";
```
Comment (NC-001) [inline NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs]: This should be removed

Comment (NC-001) [reply NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs]: The line with "TODO" :)

**Review by NC-001 — APPROVED** [review]

File: NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs [hunk 2]
```diff
@@ -19,6 +19,8 @@
 using UserNotifications;
 using static NDB.Covid19.ViewModels.InfectionStatusViewModel;
 using NDB.Covid19.iOS.Views.SelftestOption;
+using NDB.Covid19.Models;
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs:22]: Extra newline, pls delete

File: NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs [hunk 3]
```diff
@@ -47,6 +49,30 @@ public static UINavigationController GetInfectionSatusPageControllerInNavigation
             return navigationController;
         }
 
+        private void UpdateImportantMessage(ImportantMessage message)
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs]: Extra newline, pls delete

File: NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs [hunk 4]
```diff
@@ -47,6 +49,30 @@ public static UINavigationController GetInfectionSatusPageControllerInNavigation
...
+        {
+            if (message != null)
+            {
+                InformationBannerLbl.Hidden = false;
+                InformationBannerLbl.Text = message.Text;
+                InformationBannerLbl.AccessibilityLabel = message.Text;
+                InformationBannerLbl.BackgroundColor = message.BannerColor.ToUIColor();
+
+                if (message.IsClickable)
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs]: Extra newline, pls delete

File: NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs [hunk 5]
```diff
@@ -47,6 +49,30 @@ public static UINavigationController GetInfectionSatusPageControllerInNavigation
...
+                {
+                    SetupInformationBannerLink();
+                }
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs]: Extra newline, pls delete

File: NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs [hunk 6]
```diff
@@ -47,6 +49,30 @@ public static UINavigationController GetInfectionSatusPageControllerInNavigation
...
+                else
+                {
+                    InformationBannerLbl.UserInteractionEnabled = false;
+                }
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs]: Extra newline, pls delete

File: NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs [hunk 7]
```diff
@@ -75,12 +102,13 @@ public override void ViewDidLoad()
...
             }));
             MenuLabel.IsAccessibilityElement = false;
-           
+
             SetupStyling();
             MessagingCenter.Subscribe<object>(this, MessagingCenterKeys.KEY_MESSAGE_STATUS_UPDATED, OnMessageStatusChanged);
             MessagingCenter.Subscribe<object>(this, MessagingCenterKeys.KEY_APP_RETURNS_FROM_BACKGROUND, OnAppReturnsFromBackground);
             MessagingCenter.Subscribe<object>(this, MessagingCenterKeys.KEY_CONSENT_MODAL_IS_CLOSED, OnConsentModalIsClosed);
             MessagingCenter.Subscribe<object>(this, MessagingCenterKeys.KEY_UPDATE_DAILY_NUMBERS, OnAppDailyNumbersChanged);
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs]: Extra newline, pls delete

File: NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs [hunk 8]
```diff
@@ -295,6 +325,16 @@ void SetupStyling()
...
+        void SetupInformationBannerLink()
+        {
+            _bannerGestureRecognizer = new UITapGestureRecognizer();
+            _bannerGestureRecognizer.AddTarget(() => OnInformationBannerLblTapped(_bannerGestureRecognizer));
+            InformationBannerLbl.AddGestureRecognizer(_bannerGestureRecognizer);
+            InformationBannerLbl.UserInteractionEnabled = true;
+            InformationBannerLbl.AccessibilityTraits = UIAccessibilityTrait.Button;
+        }
+
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs:327]: Extra newline, pls delete

File: NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs [hunk 9]
```diff
@@ -340,6 +380,7 @@ void SetupEncounterAndInfectedButtons()
             _dailyNumbersButton = new UIButton();
             _dailyNumbersButton.TranslatesAutoresizingMaskIntoConstraints = false;
             StyleUtil.EmbedViewInsideButton(DailyNumbersView, _dailyNumbersButton);
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs]: Extra newline, pls delete

File: NDB.Covid19/NDB.Covid19/Locales/nb.json [hunk 10]
```diff
@@ -248,6 +248,7 @@
   "INFECTION_STATUS_BACKGROUND_ACTIVITY_DIALOG_MESSAGE_PART3": "Tillater du å slå den på?",
   "INFECTION_STATUS_SURVEY_HEADER_TEXT": "Spørreundersøkelse for deg som har testet deg",
   "INFECTION_STATUS_SURVEY_LINK_URL": "https://nettskjema.no/a/195329",
+  "INFECTION_STATUS_INFORMATION_BANNER_LINK_URL": "https://www.helsenorge.no/",
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19/Locales/nb.json]: I think maybe it makes sense to delete this at all? This change will either way be overwritten in the future, so maybe set the link currently directly in the ViewModel

**Review by pavshr — CHANGES_REQUESTED** [review]

File: NDB.Covid19/NDB.Covid19/Locales/nb.json (hunk 10 above)
Comment (Juliehoejrisp) [reply NDB.Covid19/NDB.Covid19/Locales/nb.json]: okay i put it in the viewmodel instead.

File: NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs [hunk 11]
```diff
@@ -47,6 +48,29 @@ public static UINavigationController GetInfectionSatusPageControllerInNavigation
...
+        {
+            if (message != null)
+            {
+                InformationBannerLbl.Hidden = false;
+                InformationBannerLbl.Text = message.Text;
+                InformationBannerLbl.AccessibilityLabel = message.Text;
+                InformationBannerLbl.BackgroundColor = message.BannerColor.ToUIColor();
+
+                if (message.IsClickable)
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs]: Extra line, remove it

File: NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs [hunk 12]
```diff
@@ -47,6 +48,29 @@ public static UINavigationController GetInfectionSatusPageControllerInNavigation
...
+                {
+                    SetupInformationBannerLink();
+                }
+                else
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs]: Extra line, remove it

File: NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs [hunk 13]
```diff
@@ -131,17 +156,18 @@ private void OnConsentModalIsClosed(object obj)
...
             SetPermissionManager();
-            _viewModel.NewMessagesIconVisibilityChanged += OnNewMessagesIconVisibilityChanged;
-
+            _viewModel.NewMessagesIconVisibilityChanged += OnNewMessagesIconVisibilityChanged;            
             _dailyNumbersButton.TouchUpInside += OnDailyNumbersBtnTapped;
             _messageViewBtn.TouchUpInside += OnMessageBtnTapped;
             _areYouInfectedBtn.TouchUpInside += OnAreYouInfectedBtnTapped;
 
+         
+
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/InfectionStatus/InfectionStatusViewController.cs]: Extra lines, remove them

**Review by pavshr — CHANGES_REQUESTED** [review]

**Review by pavshr — APPROVED** [review]

**Review by pavshr — APPROVED** [review]

File: NDB.Covid19/NDB.Covid19/ViewModels/InfectionStatusViewModel.cs [hunk 14]
```diff
@@ -43,6 +43,7 @@ public class InfectionStatusViewModel
         public static string INFECTION_STATUS_DAILY_NUMBERS_LAST_UPDATED_ACCESSIBILITY_TEXT => "INFECTION_STATUS_DAILY_NUMBERS_LAST_UPDATED_ACCESSIBILITY_TEXT".Translate();
         public static string INFECTION_STATUS_SURVEY_HEADER_TEXT => "INFECTION_STATUS_SURVEY_HEADER_TEXT".Translate();
         public static string INFECTION_STATUS_SURVEY_LINK_URL => "INFECTION_STATUS_SURVEY_LINK_URL".Translate();
+        public static string INFECTION_STATUS_INFORMATION_BANNER_LINK_URL => "https://www.helsenorge.no/".Translate();
```
Comment (NC-001) [inline NDB.Covid19/NDB.Covid19/ViewModels/InfectionStatusViewModel.cs:46]: Please remove ".Translate()" after the link

**Pull request merged**